pkg/gatherers/conditional/conditions.go

```diff
@@ -100,14 +100,19 @@ func (g *Gatherer) doesClusterVersionMatch(expectedVersionExpression string) (bo
 		return false, err
 	}
 
-	expectedRange, err := semver.ParseRange(expectedVersionExpression)
+	shortClusterVersion, err := semver.Parse(g.clusterVersion)
 	if err != nil {
 		return false, err
 	}
 
-	// ignore everything after the first three numbers
-	clusterVersion.Pre = nil
-	clusterVersion.Build = nil
+	// ignore everything after the first three numbers for the short version
```

Comment:
I am not sure I fully understand. Does it mean please that we ignore z-stream version (e.g 4.10.5 simply falls under 4.10?) ? Maybe it's OK (but maybe not :)) but we should cover it in the tests IMO.

Reply:
No, in the version `4.10.5-0.nightly-2022-05-25-193227` we ignore `-0.nightly-2022-05-25-193227` in case the version in the condition doesn't use it, for example `>4.10.5`. If the version in the condition does use it, for example the condition is `>4.10.5-0.nightly-2022-05-25-193227`, then it's used.

Reply:
I'll add more tests

Reply:
OK great. Thank you